### PR TITLE
LOG-5977: Update the API to use kube standards

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* updates the API to use kube standards
* makes ES version required and not defaulted

### Links
https://issues.redhat.com/browse/LOG-5977

cc @cahartma @xperimental @Clee2691 @libander @anpingli

### CODE REVIEW

**Review by xperimental — COMMENTED** [review]
Have gone through all the fields and checked for inconsistencies (mainly looked at the JSON/YAML representation, because that will be harder to change in the future).

Found the following inconsistencies with the new scheme (can't provide suggestions directly on the files though) and some other ideas that crossed my mind:

- `Managed` and `Unmanaged` - should we rename to `managed` and `unmanaged` as well?
- `inputRefs` - "refs" is an abbeviation. If we want to avoid this, maybe `inputNames` is an alternative? (similarly `outputRefs`, `filterRefs`)
- `inputsStatus` - "looks wrong". I think my suggestion would be `inputConditions`. Similarly for outputs, filters and pipelines.
- `kubeAPI`, `openshiftAPI` - these seem to be forgotten as enum values for `AuditSource`
- `RFC3164`, `RFC5424` - should we lower-case them to make them "less loud"?
- `delivery` - Has nothing to do with the standard, but for me `deliveryMode` looks nicer. This is a weak recommendation though.

**Comment by jcantrill** [comment]
> Have gone through all the fields and checked for inconsistencies (mainly looked at the JSON/YAML representation, because that will be harder to change in the future).
> 
> Found the following inconsistencies with the new scheme (can't provide suggestions directly on the files though) and some other ideas that crossed my mind:
> 
> * `Managed` and `Unmanaged` - should we rename to `managed` and `unmanaged` as well? 

This is a hold over from the earlier release.  Personally, I never liked capitalized values and there doesn't seem to be any guidance regarding it.  To be consistent with other enums, however, I like this change :+1: 

> * `inputRefs` - "refs" is an abbeviation. If we want to avoid this, maybe `inputNames` is an alternative? (similarly `outputRefs`, `filterRefs`)

We had an opportunity to change this in the initial review and I would recommend leave as-is given where we are.  :-1: 

> * `inputsStatus` - "looks wrong". I think my suggestion would be `inputConditions`. Similarly for outputs, filters and pipelines.

This has bothered me as well but I would propose an alternate of just "input".  Though, one could argue by calling it "InputConditions" it better fits with the other status fields :+1: 

> * `kubeAPI`, `openshiftAPI` - these seem to be forgotten as enum values for `AuditSource`

I intentionally left these as 'openshiftApi' looks wrong to me.  "Where used as a constant, all letters should be uppercase, such as "TCP" or "UDP"."  :-1: 

> * `RFC3164`, `RFC5424` - should we lower-case them to make them "less loud"?

I can get onboard with this though "rfc" also is well known acronym :+1: 

> * `delivery` - Has nothing to do with the standard, but for me `deliveryMode` looks nicer. This is a weak recommendation though.

:+1:

File: api/observability/v1/output_types.go [hunk 1]
```diff
@@ -409,13 +409,13 @@ type CloudwatchIAMRole struct {
 }
```
[the remark this answers is not in the document]
Comment (jcantrill) [reply api/observability/v1/output_types.go:409]: I purposely chose to leave as-is since ARN is all over the cloudwatch docs and is a recognized acronym

**Comment by jcantrill** [comment]
@xperimental pushed latest changes per my previous comment

File: api/observability/v1/filter_types.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ type FilterSpec struct {
 
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Kubernetes API Audit Filter"
-	KubeAPIAudit *KubeAPIAudit `json:"kubeAPIAudit,omitempty"`
+	KubeApiAudit *KubeApiAudit `json:"kubeApiAudit,omitempty"`
```
Comment (xperimental) [inline api/observability/v1/filter_types.go:73]: Need to update the validation in `FilterSpec` mentioning `kubeAPIAudit`.

File: api/observability/v1/filter_types.go [hunk 3]
```diff
@@ -91,7 +91,7 @@ type FilterSpec struct {
 	//
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Labels"
-	OpenShiftLabels map[string]string `json:"openShiftLabels,omitempty"`
+	OpenshiftLabels map[string]string `json:"openshiftLabels,omitempty"`
```
Comment (xperimental) [inline api/observability/v1/filter_types.go:94]: Need to update the validation in `FilterSpec` mentioning `openShiftLabels`.

File: test/functional/outputs/kafka/forward_to_kafka_test.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/openshift/cluster-logging-operator/test/helpers/kafka"
 )
 
-var _ = Describe("[Functional][Outputs][Kafka] Functional tests", func() {
+var _ = Describe("[Functional][OutputConditions][Kafka] Functional tests", func() {
```
Comment (xperimental) [inline test/functional/outputs/kafka/forward_to_kafka_test.go:17]: I think this needs to be rolled back. Similarly the other `[OutputConditions]` strings in the tests?

**Review by xperimental — COMMENTED** [review]
Left some comments. When I made the additional changes to the CRD it again deployed fine. A test-deployment of CLF also worked :+1:

File: test/functional/outputs/kafka/forward_to_kafka_test.go [hunk 5]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/openshift/cluster-logging-operator/test/helpers/kafka"
 )
 
-var _ = Describe("[Functional][Outputs][Kafka] Functional tests", func() {
+var _ = Describe("[Functional][Output][Kafka] Functional tests", func() {
```
Comment (xperimental) [inline test/functional/outputs/kafka/forward_to_kafka_test.go:17]:
```suggestion
var _ = Describe("[Functional][Outputs][Kafka] Functional tests", func() {
```

File: test/functional/outputs/logstash/forward_to_logstash_test.go [hunk 6]
```diff
@@ -17,7 +17,7 @@ import (
 	"github.com/openshift/cluster-logging-operator/test/matchers"
 )
 
-var _ = Describe("[Functional][Outputs][Logstash] Output to Logstash", func() {
+var _ = Describe("[Functional][OutputConditions][Logstash] Output to Logstash", func() {
```
Comment (xperimental) [inline test/functional/outputs/logstash/forward_to_logstash_test.go:20]:
```suggestion
var _ = Describe("[Functional][Outputs][Logstash] Output to Logstash", func() {
```

File: test/functional/outputs/loki/application_logs_vector_test.go [hunk 7]
```diff
@@ -15,7 +15,7 @@ import (
 	"github.com/openshift/cluster-logging-operator/test/helpers/loki"
 )
 
-var _ = Describe("[Functional][Outputs][Loki] Forwarding to Loki", func() {
+var _ = Describe("[Functional][OutputConditions][Loki] Forwarding to Loki", func() {
```
Comment (xperimental) [inline test/functional/outputs/loki/application_logs_vector_test.go:18]:
```suggestion
var _ = Describe("[Functional][Outputs][Loki] Forwarding to Loki", func() {
```

File: test/functional/outputs/loki/audit_logs_vector_test.go [hunk 8]
```diff
@@ -13,7 +13,7 @@ import (
 	obstestruntime "github.com/openshift/cluster-logging-operator/test/runtime/observability"
 )
 
-var _ = Describe("[Functional][Outputs][Loki] Forwarding to Loki", func() {
+var _ = Describe("[Functional][OutputConditions][Loki] Forwarding to Loki", func() {
```
Comment (xperimental) [inline test/functional/outputs/loki/audit_logs_vector_test.go:16]:
```suggestion
var _ = Describe("[Functional][Outputs][Loki] Forwarding to Loki", func() {
```

File: test/functional/outputs/multiple/forward_to_multiple_outputs_test.go [hunk 9]
```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/openshift/cluster-logging-operator/test/framework/functional"
 )
 
-var _ = Describe("[Functional][Outputs][Multiple]", func() {
+var _ = Describe("[Functional][OutputConditions][Multiple]", func() {
```
Comment (xperimental) [inline test/functional/outputs/multiple/forward_to_multiple_outputs_test.go:15]:
```suggestion
var _ = Describe("[Functional][Outputs][Multiple]", func() {
```

File: test/functional/outputs/multiple/multiple_test.go [hunk 10]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/openshift/cluster-logging-operator/test/framework/functional"
 )
 
-var _ = Describe("[Functional][Outputs][Multiple] tests", func() {
+var _ = Describe("[Functional][OutputConditions][Multiple] tests", func() {
```
Comment (xperimental) [inline test/functional/outputs/multiple/multiple_test.go:17]:
```suggestion
var _ = Describe("[Functional][Outputs][Multiple] tests", func() {
```

File: test/functional/outputs/multiple/suite_test.go [hunk 11]
```diff
@@ -9,5 +9,5 @@ import (
 
 func TestFunctionalOutputs(t *testing.T) {
 	RegisterFailHandler(Fail)
-	RunSpecs(t, "[Functional][Outputs][Multiple] Test Suite")
+	RunSpecs(t, "[Functional][OutputConditions][Multiple] Test Suite")
```
Comment (xperimental) [inline test/functional/outputs/multiple/suite_test.go:12]:
```suggestion
	RunSpecs(t, "[Functional][Outputs][Multiple] Test Suite")
```

File: test/functional/outputs/otlp/forward_to_otlp_test.go [hunk 12]
```diff
@@ -17,7 +17,7 @@ const (
 	timestampNano = "1693227568573159188"
 )
 
-var _ = Describe("[Functional][Outputs][OTLP] Functional tests", func() {
+var _ = Describe("[Functional][OutputConditions][OTLP] Functional tests", func() {
```
Comment (xperimental) [inline test/functional/outputs/otlp/forward_to_otlp_test.go:20]:
```suggestion
var _ = Describe("[Functional][Outputs][OTLP] Functional tests", func() {
```

File: test/functional/outputs/splunk/suite_test.go [hunk 13]
```diff
@@ -9,5 +9,5 @@ import (
 
 func TestFunctionalOutputSplunk(t *testing.T) {
 	RegisterFailHandler(Fail)
-	RunSpecs(t, "[Functional][Outputs][Splunk] Suite")
+	RunSpecs(t, "[Functional][OutputConditions][Splunk] Suite")
```
Comment (xperimental) [inline test/functional/outputs/splunk/suite_test.go:12]:
```suggestion
	RunSpecs(t, "[Functional][Outputs][Splunk] Suite")
```

File: test/functional/outputs/syslog/forward_to_syslog_test.go [hunk 14]
```diff
@@ -17,7 +17,7 @@ import (
 	"github.com/openshift/cluster-logging-operator/test/framework/functional"
 )
 
-var _ = Describe("[Functional][Outputs][Syslog] Functional tests", func() {
+var _ = Describe("[Functional][OutputConditions][Syslog] Functional tests", func() {
```
Comment (xperimental) [inline test/functional/outputs/syslog/forward_to_syslog_test.go:20]:
```suggestion
var _ = Describe("[Functional][Outputs][Syslog] Functional tests", func() {
```

File: test/functional/outputs/syslog/rfc3164_test.go [hunk 15]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/openshift/cluster-logging-operator/test/framework/functional"
 )
 
-var _ = Describe("[Functional][Outputs][Syslog] RFC3164 tests", func() {
+var _ = Describe("[Functional][OutputConditions][Syslog] RFC3164 tests", func() {
```
Comment (xperimental) [inline test/functional/outputs/syslog/rfc3164_test.go:17]:
```suggestion
var _ = Describe("[Functional][Outputs][Syslog] RFC3164 tests", func() {
```

File: test/functional/outputs/syslog/rfc5424_test.go [hunk 16]
```diff
@@ -15,7 +15,7 @@ import (
 	"github.com/openshift/cluster-logging-operator/test/framework/functional"
 )
 
-var _ = Describe("[Functional][Outputs][Syslog] RFC5424 tests", func() {
+var _ = Describe("[Functional][OutputConditions][Syslog] RFC5424 tests", func() {
```
Comment (xperimental) [inline test/functional/outputs/syslog/rfc5424_test.go:18]:
```suggestion
var _ = Describe("[Functional][Outputs][Syslog] RFC5424 tests", func() {
```

File: test/functional/outputs/syslog/suite_test.go [hunk 17]
```diff
@@ -11,7 +11,7 @@ import (
 func TestFunctionalOutputs(t *testing.T) {
 	RegisterFailHandler(Fail)
 
-	tc := "[Functional][Outputs][Syslog] Test Suite"
+	tc := "[Functional][OutputConditions][Syslog] Test Suite"
```
Comment (xperimental) [inline test/functional/outputs/syslog/suite_test.go:14]:
```suggestion
	tc := "[Functional][Outputs][Syslog] Test Suite"
```

File: test/functional/outputs/unavailable_forward_output_test.go [hunk 18]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/openshift/cluster-logging-operator/test/helpers/oc"
 )
 
-var _ = Describe("[Functional][Outputs][Unavailable] Output", func() {
+var _ = Describe("[Functional][OutputConditions][Unavailable] Output", func() {
```
Comment (xperimental) [inline test/functional/outputs/unavailable_forward_output_test.go:17]:
```suggestion
var _ = Describe("[Functional][Outputs][Unavailable] Output", func() {
```

File: internal/generator/vector/output/config_strategy.go [hunk 19]
```diff
@@ -46,7 +46,7 @@ func (o Output) VisitRequest(r common.Request) common.Request {
 // VisitBuffer modifies the buffer behavior depending upon the value
 // of the tuning.Delivery mode
 func (o Output) VisitBuffer(b common.Buffer) common.Buffer {
-	switch o.tuning.Delivery {
+	switch o.tuning.DeliveryMode {
```
Comment (xperimental) [inline internal/generator/vector/output/config_strategy.go:49]: The same change also needs to be made in `test/helpers/outputs/adapter/fake/config_strategy.go`.

File: internal/generator/vector/output/config_strategy_test.go [hunk 20]
```diff
@@ -21,7 +21,7 @@ func (s *fakeSink) SetCompression(algo string) {
 	s.Compression = algo
 }
 
-var _ = Describe("ConfigStrategy for tuning Outputs", func() {
+var _ = Describe("ConfigStrategy for tuning OutputConditions", func() {
```
Comment (xperimental) [inline internal/generator/vector/output/config_strategy_test.go:24]:
```suggestion
var _ = Describe("ConfigStrategy for tuning Outputs", func() {
```

File: internal/validations/observability/filters/validate_filters_test.go [hunk 21]
```diff
@@ -8,7 +8,7 @@ import (
 	. "github.com/openshift/cluster-logging-operator/test/matchers"
 )
 
-var _ = Describe("[internal][validations] ClusterLogForwarder: Filters", func() {
+var _ = Describe("[internal][validations] ClusterLogForwarder: FilterConditions", func() {
```
Comment (xperimental) [inline internal/validations/observability/filters/validate_filters_test.go:11]:
```suggestion
var _ = Describe("[internal][validations] ClusterLogForwarder: Filters", func() {
```

**Pull request merged**